code/modules/mob/living/simple_animal/friendly/mouse.dm

```diff
@@ -124,5 +129,18 @@
 	else
 		return ..()
 
+/obj/item/reagent_containers/food/snacks/deadmouse/afterattack(obj/target, mob/living/user, proximity_flag)
+	if(proximity_flag && reagents && target.is_open_container())
+		// is_open_container will not return truthy if target.reagents doesn't exist
+		var/datum/reagents/target_reagents = target.reagents
+		var/fuel_amount = reagents.maximum_volume - reagents.total_volume * (4 / 3)
+		if(target_reagents.has_reagent(/datum/reagent/fuel) && target_reagents.trans_id_to(src, /datum/reagent/fuel, fuel_amount))
```

Comment:
I kinda dislike how this only transfers welding fuel and nothing else.
May I ask why you are not just doing:
if(target_reagents.trans_to(src, fuel_amount, transfered_by = user))

Reply:
It will mean that any chem that isn't organic slurry will be stuck in the mouse and you won't be able to use that mouse if it's all clogged up. If I trans_to back out of the mouse then the nutriment and vitamin already in the mouse will come too, and I don't want that.

The only thing that needs to be in the mouse is welder fuel to make the reaction happen, and it all gets transferred out immediately anyhow (if there's room). It's a little cheaty but a lot easier to keep the original nutriment and vitamin that way.

Reply:
Wouldn't you want to pull out the vita/nutrients at least for the purpose of MUH IMMERSHUN?

It's a ghetto af mechanic, I think having that small detail helps build that notion.

Reply:
In my view the vitamins and nutrients are in the flesh of the mouse, and don't come out with the welding fuel. Not attached though, i'll change it for you.